### PR TITLE
refactor: Unified Widget Resolver System

### PR DESCRIPTION
**Pull request by sudo-Tiz**
# Refactor: Unified Widget Resolver System

## 🎯 Summary
Replaced complex factory pattern with unified `WidgetResolver` that handles all widget types consistently. Eliminated code duplication and resolved circular import issues.

## 🔧 Changes
- **NEW**: `utils/widget_factory.py` - Unified resolver with single dispatch pattern
- **REFACTOR**: `utils/functions.py` - Consolidated validation logic, removed duplicate functions
- **UPDATE**: Widget containers to use unified resolver system
- **FIX**: Circular import resolution through direct validation
- **CLEAN**: JSON schema reference fix

## 📊 Impact
- **Code reduction**: ~149 → ~100 lines in widget factory
- **DRY compliance**: Eliminated duplicate validation functions
- **KISS principle**: Single resolver vs multiple factory classes
- **Performance**: Reduced object instantiation overhead

## ⚠️ Areas for Improvement
1. **Error handling**: Consider custom exception types instead of generic `ValueError`
2. **Validation caching**: Widget validation runs on every config parse
3. **Type safety**: Missing type hints in some resolver methods
4. **Testing**: No unit tests for new resolver system

## 🧪 Testing
- [x] Manual verification - all widget types resolve correctly
- [x] Circular import resolved
- [x] Lint compliance achieved
- [ ] **TODO**: Add unit tests for edge cases

## 🚀 Migration Notes
No breaking changes - existing configurations work unchanged. Internal refactoring only.

### CODE REVIEW

File: utils/widget_factory.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+from typing import Any, Dict, Optional
+
+from loguru import logger
+
+from shared.custom_button import CustomButtonWidget
+
+
+class WidgetResolver:
+    """Universal widget resolver with unified handling for all widget types."""
```
[the remark this answers is not in the document]
Comment (sudo-Tiz) [reply utils/widget_factory.py:45]: I beleive it's fixed in the new commits

File: utils/functions.py [hunk 2]
```diff
@@ -427,50 +427,103 @@ def kill_process(process_name: str):
...
-    group_idx = widget.replace(group_prefix, "", 1)
-
-    if not group_idx.isdigit():
-        raise ValueError(
-            f"Invalid {group_type} group index "
-            f"'{group_idx}' in section {section}. Must be a number."
-        )
+def validate_widget_reference(
+    widget_spec: str, parsed_data: dict, default_config: dict, section: str = "layout"
+):
```
[the remark this answers is not in the document]
Comment (sudo-Tiz) [reply utils/functions.py:526]: I beleive it's fixed in new commits
